pkg/comp-functions/functions/vshnnextcloud/deploy.go

```diff
@@ -143,6 +157,110 @@ func DeployNextcloud(ctx context.Context, comp *vshnv1.VSHNNextcloud, svc *runti
 	return nil
 }
 
+func getObservedPostgresSettings(svc *runtime.ServiceRuntime, comp *vshnv1.VSHNNextcloud) (map[string]string, map[string]string, string, error) {
+	pgBouncerConfig := map[string]string{
+		"max_client_conn": "1000",
+	}
+	pgSettings := map[string]string{
+		"max_connections": "1000",
+	}
+	pgDiskSize := ""
+
+	if !comp.Spec.Parameters.Service.UseExternalPostgreSQL {
+		return pgBouncerConfig, pgSettings, pgDiskSize, nil
+	}
+
+	existing := &vshnv1.XVSHNPostgreSQL{}
+	err := svc.GetObservedComposedResource(existing, comp.GetName()+pgInstanceNameSuffix)
+
+	switch {
+	case err == nil:
+		if existing.Spec.Parameters.Size.Disk != "" {
+			pgDiskSize = existing.Spec.Parameters.Size.Disk
+		}
+
+		if existing.Spec.Parameters.Service.PgBouncerSettings != nil {
+			if bouncer := existing.Spec.Parameters.Service.PgBouncerSettings.Pgbouncer.Raw; len(bouncer) > 0 {
+				var rawBouncer map[string]interface{}
+				if err := json.Unmarshal(bouncer, &rawBouncer); err != nil {
+					return nil, nil, "", fmt.Errorf("cannot unmarshal existing pgbouncersettings: %w", err)
+				}
+
+				bouncerSettings := make(map[string]string)
+				for k, v := range rawBouncer {
+					bouncerSettings[k] = fmt.Sprint(v)
+				}
+
+				if err := mergo.Merge(&pgBouncerConfig, bouncerSettings, mergo.WithOverride); err != nil {
+					return nil, nil, "", fmt.Errorf("cannot merge existing pgbouncerconfig: %w", err)
+				}
+			}
+		}
+
+		if settings := existing.Spec.Parameters.Service.PostgreSQLSettings.Raw; len(settings) > 0 {
+			var rawSettings map[string]interface{}
+			if err := json.Unmarshal(settings, &rawSettings); err != nil {
+				return nil, nil, "", fmt.Errorf("cannot unmarshal existing pgsettings: %w", err)
+			}
+
+			pgSettingsMap := make(map[string]string)
+			for k, v := range rawSettings {
+				pgSettingsMap[k] = fmt.Sprint(v)
+			}
+
+			if err := mergo.Merge(&pgSettings, pgSettingsMap, mergo.WithOverride); err != nil {
+				return nil, nil, "", fmt.Errorf("cannot merge existing pgsettings: %w", err)
+			}
+		}
+
+	case err == runtime.ErrNotFound:
+		pgDiskSize = "10Gi"
+
+	default:
+		return nil, nil, "", fmt.Errorf("cannot get observed postgres instance: %w", err)
+	}
+
+	if params := comp.Spec.Parameters.Service.PostgreSQLParameters; params != nil {
+		if params.Size.Disk != "" && !(pgDiskSize == "10Gi" && params.Size.Disk < pgDiskSize) {
```

Comment:
We should probably convert the strings to int for the comparison. See: https://go.dev/play/p/0Kubqu2OQ-D

For strings with the less than and greater than operator golang will look at the underlying bytes, which may or may not match the actual value.

You can use `k8s.io/apimachinery/pkg/api/resource` to parse the resource strings and compare them. See `quotas.go` for how to use it.